### PR TITLE
Add transfer button in GoalPage 

### LINKED ISSUE #1166: Add transfer button in GoalPage
image

### PR DESCRIPTION
**Pull request by awinogradov**
## PR includes

- [x] Refactor
- [x] Feature

## Related issues

Resolve #1166

## QA Instructions, Screenshots, Recordings
  
image
image

### CODE REVIEW

File: src/components/GoalForm/GoalForm.tsx [hunk 1]
```diff
@@ -38,16 +36,10 @@ interface GoalFormProps {
     validityScheme: Schema;
     id?: string;
     tip?: React.ReactNode;
-    help?: AvailableHelpPages;
 
     onSumbit: (fields: z.infer<GoalFormProps['validityScheme']>) => void;
 }
 
-const StyledHelpIcon = styled(QuestionIcon)`
```
Comment (awinogradov) [inline src/components/GoalForm/GoalForm.tsx:46]: Unused code after https://github.com/taskany-inc/issues/pull/1139

File: src/components/GoalForm/GoalForm.tsx [hunk 2]
```diff
@@ -137,19 +128,21 @@ export const GoalForm: React.FC<GoalFormProps> = ({
...
-                                <GoalParentComboBox
-                                    text={tr('Enter project or team title')}
-                                    placeholder={tr('Enter project or team title')}
-                                    error={errorsResolver(field.name)}
-                                    disabled={busy}
-                                    {...field}
-                                />
-                            )}
-                        />
+                        {!id && (
```
Comment (9teen90nine) [inline src/components/GoalForm/GoalForm.tsx:131]: nullable?

File: src/components/GoalParentComboBox.tsx [hunk 3]
```diff
@@ -32,7 +49,11 @@ export const GoalParentComboBox = React.forwardRef<HTMLDivElement, GoalParentCom
             setInputState(value?.title || query);
         }, [value, query]);
 
-        const { data } = trpc.project.suggestions.useQuery(inputState);
+        const { data } = trpc.project.suggestions.useQuery(inputState, {
+            enabled: inputState.length >= 2,
+            cacheTime: 0,
```
Comment (9teen90nine) [inline src/components/GoalParentComboBox.tsx:54]: why not use a small cacheTime as a throttle? doesn't it include the query?

File: trpc/router/goal.ts [hunk 4]
```diff
@@ -358,8 +358,44 @@ export const goal = router({
...
+                include: {
+                    ...goalDeepQuery,
+                },
+            });
+
+            return {
+                ...goal,
+                ...addCalclulatedGoalsFields(goal, activityId),
+            };
+        } catch (error: any) {
```
Comment (9teen90nine) [inline trpc/router/goal.ts:396]: I think I've seen this catch already, maybe it should be a helper function?

**Review by 9teen90nine — APPROVED** [review]

File: src/components/GoalParentComboBox.tsx (hunk 3 above)
Comment (awinogradov) [reply src/components/GoalParentComboBox.tsx:54]: it is react-query trick ¯\_(ツ)_/¯

File: trpc/router/goal.ts (hunk 4 above)
Comment (awinogradov) [reply trpc/router/goal.ts:396]: https://github.com/taskany-inc/issues/issues/1216

**Pull request merged**